### DISCUSSION TITLE
optional path parameter

### DISCUSSION THREAD

**Discussion by Swarkin** [rwf2/Rocket#2816] [Questions] [answered]
how can I do something like this:
```rust
#[get("/times/<level>/<amount>")]
fn times(level: &str, amount: Option<u32>)
```

**Comment by the10thWiz** [comment, marked as answer]
This isn't possible with the current macros. (`Option<u32>` will return `Some(...)` if the value is a valid `u32`, and `None` if it isn't a valid `u32` - e.g. `abc` would be parsed to `None`)

You will have to mount two different routes, one with and one without the final parameter. To reduce code duplication, I would recommend creating a helper that actually implements the logic, and each route simply calls the helper.

```rust
fn times(level: &str, amount: Option<u32>) { ... }

#[get("/times/<level>")]
fn times_one(level: &str) {
    times(level, None)
}
#[get("/times/<level>/<amount>")]
fn times_two(level: &str, amount: u32) {
    times(level, Some(amount))
}
```
